Testing/SystemTests/tests/analysis/INTERReductionTest.py

```diff
@@ -0,0 +1,349 @@
+        workspace_name = WorkspaceName(file)
+        Load(file, OutputWorkspace=workspace_name)
+        workspace_names.append(workspace_name)
+
+    output_workspace_name = 'Output'
+    GroupWorkspaces(InputWorkspaces=workspace_names, OutputWorkspace=output_workspace_name)
+    mtd[output_workspace_name].sortByName()
+    SaveNexus(InputWorkspace=output_workspace_name, Filename=output_filename)
+
+
+def RegenerateRunsFile(transmission_run_names, run_numbers, event_run_numbers):
+    '''Generate the test input file from a range of run numbers and transmission runs.'''
+    # Load transmission runs
+    for run in transmission_run_names:
+        Load('{}.raw'.format(run), OutputWorkspace=run)
+    # Load raw run files
+    run_names = [str(run_number)+'.raw' for run_number in run_numbers]
+    for run_name in run_names:
+        Load(run_name, OutputWorkspace=run_name)
+    # Load event workspaces
+    event_run_names = [str(event_run_number) for event_run_number in event_run_numbers]
+    for event_run_name in event_run_names:
+        LoadEventNexus(event_run_name, OutputWorkspace=event_run_name, LoadMonitors=True)
+    event_monitor_names = [str(run_number)+'_monitors' for run_number in event_run_numbers]
+    # Group and save
+    GroupWorkspaces(InputWorkspaces=run_names + transmission_run_names + event_run_names +
+                    event_monitor_names,
+                    OutputWorkspace='Input')
+    SaveNexus(InputWorkspace='Input', Filename='INTERReductionTestRuns.nxs')
+
+
+def CreateTransmissionWorkspaces(runs1, runs2, output_names):
+    for run1, run2, name in zip(runs1, runs2, output_names):
+        CreateTransmissionWorkspaceAuto(
+            FirstTransmissionRun=run1,
+            SecondTransmissionRun=run2,
+            OutputWorkspace=name,
+            StartOverlap=10,
+            EndOverlap=12)
+        # Delete debug workspaces
+        DeleteWorkspace('TRANS_LAM_'+run1)
+        DeleteWorkspace('TRANS_LAM_'+run2)
+
+
+def EventRef(run_number,angle,start=0,stop=0,DB='TRANS'):
+    '''Event data time-slicing'''
+    # Filter the input workspace by the given start/stop time (or end time
+    # if stop time is not given)
+    run_name=str(run_number)
+    run_workspace=mtd[run_name]
+    if stop==0:
+        stoptime=run_workspace.getRun().getLogData('duration').value
+    else:
+        stoptime=stop
+    filter_ws_name=run_name+'_filter'
+    FilterByTime(InputWorkspace=run_name, OutputWorkspace=filter_ws_name, StartTime=start, StopTime=stoptime)
+    # Calculate the fraction of proton charge in this slice
+    filter_workspace=mtd[filter_ws_name]
+    slice_proton_charge = filter_workspace.getRun().getLogData('gd_prtn_chrg').value
+    total_proton_charge = run_workspace.getRun().getLogData('gd_prtn_chrg').value
+    fraction = slice_proton_charge/total_proton_charge
+    duration = filter_workspace.getRun().getLogData('duration').value
+    print('Fraction:',fraction)
+    print('Slice:',slice)
```

Comment:
Is `slice` a variable in this context?

Reply:
Sorry, that got left in after it was renamed to `slice_proton_charge`. Thanks for spotting it!